llvm/tools/llvm-readobj/XCOFFDumper.cpp

```diff
@@ -983,6 +983,17 @@ const EnumEntry<XCOFF::SectionTypeFlags> SectionTypeFlagsNames[] = {
 #undef ECase
 };
 
+const EnumEntry<XCOFF::DwarfSectionSubtypeFlags>
```

Comment:
You need a dedicated test in the llvm-readobj tests that shows the llvm-readobj behaviour. The yaml2obj test doesn't need to execute llvm-readobj: just checking the YAML output from obj2yaml is probably sufficient.

Reply:
The test changes made weren't exactly what I had in mind. I was expecting to see the following:

1) A yaml2obj test that has a section with each DWARF sub-section type and shows that obj2yaml produces a YAML that contains the correct type values. No llvm-readobj involvement.
2) An llvm-readobj test that takes a yaml2obj-generated input (quite possibly identical to the one from 1), or an extended version of an existing YAML in an llvm-readobj test that has the necessary set of sections), and dumps the sections, showing that the sub-section type for each value is correct. Note that in this case, testing all the other fields is unnecessary: just the section name and sub-section type would be sufficient (testing other fields is just noise that makes the test harder to follow).